Use 1es releasejob for deployments

hallipr opened the pull request:
- Only use `deployment:` step for package deployment job
  - deployment steps should not clone the repo or download resources other than build artifacts

weshaggard commented on eng/pipelines/templates/jobs/ci.yml:
```diff
@@ -209,6 +209,10 @@ jobs:
           ignoreDirectories: '.nuget'
 
+      - pwsh: |
+          Copy-Item -Path '$(Build.SourcesDirectory)/eng/CredScanSuppression.json' -Destination `$(Build.ArtifactStagingDirectory)/CredScanSuppression.json` -Force
```
Why is this needed? I would like to avoid this.

hallipr replied:
removed

weshaggard commented on eng/pipelines/templates/stages/archetype-net-release.yml:
```diff
@@ -117,31 +111,41 @@ stages:
                     deploy:
                       steps:
-                        - template: /eng/pipelines/templates/steps/download-credscan-suppressions.yml
+                        - template: /eng/pipelines/templates/steps/copy-credscan-suppressions.yml
```
You shouldn't need this for deployment

hallipr replied:
removed

scbedd commented on eng/pipelines/templates/stages/archetype-net-release.yml:
```diff
@@ -117,31 +111,37 @@ stages:
-                  outputParentDirectory: '$(Pipeline.Workspace)'
-                  outputs:
-                    - output: nuget
-                      displayName: 'Publish ${{artifact.name}} package to NuGet.org'
```
I'm a little bit weirded out by the output artifacts used here, as historically I believe they were added to reduce the amount of generated YML. We should try a build of `storage` or `comunication` release (just up to triggering the release) to ensure that we don't hit max yml size without having this output inlined in the templatecontext.

hallipr replied:
The output artifacts allow you to compress your injected SDL steps by combining all of the scans across all of your outputs. This is in contract to inline output tasks like `1ES.PublishNuget` or `1ES.PublishArtifact` that will inject all of the sdl steps per inlined task.

hallipr replied:
The amount of YML doesn't really change.  We were declaring 2 job outputs per artifact, and now we're declaring 2 publish tasks per artifact.

scbedd commented on eng/pipelines/templates/stages/1es-redirect.yml:
```diff
@@ -11,7 +11,7 @@ resources:
       type: git
       name: internal/azure-sdk-build-tools
-      ref: refs/tags/azure-sdk-build-tools_20250228.1
+      ref: refs/heads/users/pahallis/credscan
```
[FYI for reviewers](https://dev.azure.com/azure-sdk/internal/_git/azure-sdk-build-tools/pullrequest/1156)

scbedd replied:
Does the credscansuppression file coming down as part of azure-sdk-build-tools put the credscan suppression file in the same place as exists in the repo? I would expect it to exist under a nonstandard path `$(Pipeline.Workspace)/azure-sdk-build-tools/eng/CredScanSuppression.json`.

hallipr replied:
Path variables aren't consistent when mixed with different `checkout` step combinations.

When a run has a single `checkout` step, `$(System.DefaultWorkingDirectory)` always points to the single repo path, even when the repo uses a custom path.

When there are multiple `checkout` steps, `$(System.DefaultWorkingDirectory)` always points to `$(Pipeline.Workspace)/s`

The credscan path seems to be rooted with `$(System.DefaultWorkingDirectory)`, so it should track single repo checkouts, but the sdl settings will need to be overriden for any multi-repo checkout.

I added a wiki page that covers how the path variables are affected by different repo counts and paths:
https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/1222/Pipeline-Path-Variables

hallipr commented on eng/pipelines/templates/stages/archetype-net-release.yml:
```diff
@@ -27,20 +26,19 @@ stages:
+          - download: current
+            artifact: ${{parameters.ArtifactName}}
+            timeoutInMinutes: 5
+          - template: /eng/pipelines/templates/steps/copy-credscan-suppressions.yml
```
switched to sparse checkout

scbedd commented:
/azp run net - core - ci

weshaggard commented on eng/pipelines/templates/stages/archetype-net-release.yml:
```diff
@@ -156,11 +156,9 @@ stages:
-                      parameters:
-                        SkipCheckoutNone: true
+                    - checkout: self
+                      fetchDepth: 1
```
Why this change?

hallipr replied:
reverted to sparse checkout with explicit path

weshaggard commented on eng/pipelines/templates/stages/archetype-net-release.yml:
```diff
@@ -153,120 +151,113 @@ stages:
                   dependsOn: PublishPackage
 
                   pool:
                     name: azsdk-pool-mms-ubuntu-2004-general
```
As a follow-up use image.yml variables for these.

hallipr replied:
Had more yaml work to do, so added in this PR

hallipr commented:
/azp run net - template

hallipr committed:
Use 1es releasejob for deployments

- Only use deployment: step for package deployment job
  - deployment steps should not clone the repo or download resources other than build artifacts

hallipr commented:
/check-enforcer override

Release smoke test failures aren't related to the jobs change
